src/main/java/de/hysky/skyblocker/skyblock/dungeon/GoldorWaypointsManager.java

```diff
@@ -0,0 +1,252 @@
+                } else if ((playerName = getPlayerName(LEVER_ACTIVATED.matcher(message))) != null) {
+                    RemoveNearestWaypoint(LEVERS, playerName);
+                } else if (message.equals(CORE_ENTRANCE)) {
+                    active = false;
+                }
+            }
+        } else {
+            if (message.equals(TERMINALS_START)) {
+                enableAll(TERMINALS);
+                enableAll(DEVICES);
+                enableAll(LEVERS);
+                active = true;
+            }
+        }
+    }
+
+    private static void render(WorldRenderContext context) {
+        if (active) {
+            TERMINALS.stream().filter(waypoint -> waypoint.phase == currentPhase && waypoint.shouldRender()).forEach(waypoint -> waypoint.render(context));
+            DEVICES.stream().filter(waypoint -> waypoint.phase == currentPhase && waypoint.shouldRender()).forEach(waypoint -> waypoint.render(context));
+            LEVERS.stream().filter(waypoint -> waypoint.phase == currentPhase && waypoint.shouldRender()).forEach(waypoint -> waypoint.render(context));
+        }
+    }
+
+    private static class GoldorWaypoint extends NamedWaypoint {
+        public static final Codec<GoldorWaypoint> CODEC = RecordCodecBuilder.create(i -> i.group(
+                WaypointTargetKind.CODEC.fieldOf("kind").forGetter(w -> w.kind),
+                Codec.INT.fieldOf("phase").forGetter(customWaypoint -> customWaypoint.phase),
+                TextCodecs.CODEC.fieldOf("name").forGetter(NamedWaypoint::getName),
+                BlockPos.CODEC.fieldOf("pos").forGetter(customWaypoint -> customWaypoint.pos)
+        ).apply(i, GoldorWaypoint::new));
+
+        private static final Supplier<Type> TYPE_SUPPLIER = () -> SkyblockerConfigManager.get().dungeons.goldor.waypointType;
+
+        final WaypointTargetKind kind;
+        final int phase;
+
+        GoldorWaypoint(WaypointTargetKind kind, int phase, Text name, BlockPos pos) {
+            super(pos, name, TYPE_SUPPLIER, kind.colorComponents, 0.25F, true);
+            this.kind = kind;
+            this.phase = phase;
+        }
+
+        /**
+         * The different classes of waypoints
+         */
+        enum WaypointTargetKind implements StringIdentifiable {
+            TERMINAL("terminal", 0, 255, 0),
+            DEVICE("device", 0, 0, 255),
+            LEVER("lever", 255, 255, 0);
+
+            private static final Codec<WaypointTargetKind> CODEC = StringIdentifiable.createCodec(WaypointTargetKind::values);
+            private final String name;
+            private final float[] colorComponents;
+
+            WaypointTargetKind(String name, int r, int g, int b) {
+                this.name = name;
+                this.colorComponents = new float[]{r / 255F, g / 255F, b / 255F};
+            }
+
+            @Override
+            public String asString() {
+                return this.name;
+            }
+
+            @Override
+            public String toString() {
```

Comment:
I don't see this being called so you can likely just not override it (and drop the name field too following with above comment)